### ISSUE TITLE
Add support for a border below section headers in Sidebar and Panel

### ISSUE THREAD

**Issue by a-stewart** [microsoft/vscode#157343]
For a theme that I have been trying to implement based on mockups, it would be good if we could provide a border below the section headers in the Sidebar and Panel, as well as just the separators.

Eg:  
![image](https://user-images.githubusercontent.com/150152/183220268-e08f3081-72a1-433d-8be8-cc3eb04f86f1.png)

This is a request in addition to #157312, although #157312 makes sense a standalone change, I think this change only really works well in addition to that one.

**Comment by daviddossett** [comment]
To me this adds quite a lot of business to the panels and views, even outside of a high contrast theme. Is there a functional reason you want to add these or is it just because they were in the mocks? 

Note that these would also cause issues for the focused state of a view and its elements e.g. 
![CleanShot 2022-08-08 at 09 37 38](https://user-images.githubusercontent.com/25163139/183468821-171c8d05-4263-4172-adb5-449e4cdf843e.gif)

**Comment by a-stewart** [comment]
Yeah, I would definitely propose this being an option and not the default.

Here is the theme we are working on with the borders above and below:

![image](https://user-images.githubusercontent.com/150152/183619592-764b4c48-3a94-4164-a409-57e4e2b30ade.png)

At the moment we can get close to that, but the single border on top of the panel headers makes it look at bit odd:

![image](https://user-images.githubusercontent.com/150152/183620008-f957c99d-6f0d-4e6b-bef2-7011e4f151cc.png)

This is not something that I think all themes will want to use, but it is something that would be nice to have when it is wanted.

It does seem to work with the focus state too without issue:

![image](https://user-images.githubusercontent.com/150152/183628986-0a49154b-13bc-4acb-a5bc-347f6282785d.png)

![image](https://user-images.githubusercontent.com/150152/183629148-23b9797e-4aa7-4222-a6ad-6b8993381747.png)

**Comment by albertelo** [comment]
Some more context as to why we took this approach:
internally one of the feedback that has been the most consistent across our tools has been the lack of ability to differentiate content. When users of our tools claim a UI is "too white", it normally refers to the lack of differentiation between primary and secondary content. Color alone can work, but in this case we explored adding more dividers to make it easier to differentiate. That said, we will tone down the use of dividers seen in the mocks Anthony posted, but the dividers in the bottom panel would still be useful.

**Comment by a-stewart** [comment]
Hi,

Sorry for the delay - I was on vacation and then had a bit of catching up to do once I got back.

This PR was designed to be in addition to: https://github.com/microsoft/vscode/issues/157312  
That PR adds a border between the top view and the panel / sidebar header. This PR aims to add more borders to the other views.

Since I can image there being a case for https://github.com/microsoft/vscode/issues/157312 being applied in isolation, I had done that as a separate PR with this one being an addition. But if you prefer I could modify it so that that both are included in a single setting?
